CMU-313/spring23-nodebb-team-dave
# Career model integration

### Description ###

- adds a new recruiter user type
- adds a new career tab on NodeBB, that leads to a new career page
- the new career page will 
  - allow students to submit relevant information about themselves for recruiters to see
  - allow recruiters to see a list of students who have submitted their information, together with the prediction of their success in industry

### Motivation ###
> a significant number of job recruiters have taken interest in your project and would like to use the platform as a means to reach out to students for career opportunities. However, the population of students is too large, and increasing annually.

> They are concerned by the sheer number of students to review for job applications, and recruiters want a simpler way to narrow the applicant pool.

> So we are introducing an extra ML-based feature to take in student performance data and provide a prediction of which applicants are more likely to succeed in industry

#### Model ####

- [x] Implement a new account type "recruiter" to be used during registration added to `src/controllers/index.js`
- [x] Add storage/retrieval of user-based career-related information in `src/user/career.js`
#### View/Controller ####

- [x] Adding a career page for students and recruiters to view/submit application information, all under `themes/nodebb-theme-persona/templates`
  - Student view: `themes/.../templates/partials/career/student.tpl`
  - Recruiter view: `themes/.../templates/partials/career/recruiter.tpl`
  - Career page logic: `src/controllers/career.js`


#### ML Work ####

- [x] Trained model and working starter code in career-model folder. Check its README for more information
- [x] Initial framework for integration ML recommender in src/controllers/write/career.js. Currently generates a dummy (random) value for recommendation
- [x] Complete integration of model with NodeBB codebase
- [x] Dockerize the current feature implementation microservice

### Testing ###
- make new users under student and recruiter type
- login as the student > go to careers page > fill out the info form and submit
- login as a recruiter > go to careers page > see the student entries w/ predictions
